Orange/widgets/visualize/owpythagorastree.py

```diff
@@ -0,0 +1,387 @@
+        self.view.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.view.setRenderHint(QtGui.QPainter.Antialiasing)
+
+        self.mainArea.layout().addWidget(self.view)
+
+    def set_ctree(self, ctree=None):
+        self.clear()
+        self.raw_tree = ctree
+        if ctree is not None:
+            self.tree = SklTreeAdapter(ctree.skl_model.tree_)
+
+        self._update()
+
+    def clear(self):
+        self.raw_tree = None
+        self.tree = None
+        self._clear_scene()
+
+    def _clear_scene(self):
+        self.scene.clear()
+
+    def _update(self):
+        if self.tree is not None:
+            tree_builder = PythagorasTree()
+
+            def draw_square_rec(tree_node):
+                self.scene.addItem(SquareGraphicsItem(*tree_node.square))
+                for child in tree_node.children:
+                    draw_square_rec(child)
+
+            draw_square_rec(tree_builder.pythagoras_tree(self.tree))
+        else:
+            pass
+
+    def onDeleteWidget(self):
+        self.clear()
+        super().onDeleteWidget()
+
+
+class SquareGraphicsItem(QtGui.QGraphicsRectItem):
+    """Square Graphics Item.
+
+    Square component to draw as components for the Pythagoras tree. It is
+    initialized with a given center QPointF, the main side length, and the
+    initial angle the sqaure is to be rendered with.
+
+    Parameters
+    ----------
+    center : Point
+        The central point of the square
+    length : float
+        The length of the square sides
+    angle : float
+        The initial angle at which the square will be rotated (in rads)
+
+    Attributes
+    ----------
+    center
+    length
+    angle
+
+    """
+
+    def __init__(self, center, length, angle, parent=None):
+        self._center_point = center
+        self.center = QtCore.QPointF(*center)
+        self.length = length
+        self.angle = angle
+        super().__init__(self._get_rect_attributes(), parent)
+        # no need to perform extra transformations where they won't be seen
+        if angle % 90 != 0:
+            self.setTransformOriginPoint(self.boundingRect().center())
+            self.setRotation(math.degrees(angle))
+
+    def _get_rect_attributes(self):
+        """
+        Compute the QRectF that a QGraphicsRect needs to be rendered with the
+        data passed down in the constructor.
+        """
+        height = width = self.length
+        x = self.center.x() - self.length / 2
+        y = self.center.y() - self.length / 2
+        return QtCore.QRectF(x, y, height, width)
+
+
+class TreeNode:
+    """A node in the tree structure used to represent the tree adapter
+
+    Parameters
+    ----------
+    square : Square
+        The square the represents the tree node.
+    parent : TreeNode
+        The parent of the current node.
+    children : list of TreeNode, optional
+        All the children that belong to this node.
+
+    """
+    def __init__(self, square, parent, children=[]):
```

Comment:
If default `children` has to be an iterable, should be a read-only type, e.g. tuple.